bin/centreon-sanitize-images.php

```diff
@@ -0,0 +1,316 @@
+    foreach ($files['svgImages'] as $svgImage) {
+        $pattern = str_replace('/', '\/', __DIR__ . '/../www/img/media/');
+        echo preg_replace('/' . $pattern . '/', '', $svgImage) . PHP_EOL;
+    }
+    echo PHP_EOL;
+}
+
+if (!empty($files['invalidImages'])) {
+    echo "The following images have an invalid MIME type or a mismatch between MIME type and " .
+    "file extension and can be replaced by a generic image:" . PHP_EOL;
+    foreach ($files['invalidImages'] as $invalidImg) {
+        $pattern = str_replace('/', '\/', __DIR__ . '/../www/img/media/');
+        echo preg_replace('/' . $pattern . '/', '', $invalidImg) . PHP_EOL;
+    }
+    echo PHP_EOL;
+}
+
+$proceed = askQuestion('Would you like to proceed to sanitize ? [y/N]: ');
+if (strtolower($proceed) === 'y') {
+    foreach ($files['svgImages'] as $svgImage) {
+        try {
+            sanitizeSvg($svgImage);
+        } catch (\Exception $ex) {
+            echo "ERROR - " . $ex->getMessage();
+            exit(1);
+        }
+    }
+
+    foreach ($files['invalidImages'] as $invalidImg) {
+        try {
+            convertCorruptedImageOrFail($invalidImg);
+        } catch (\Exception $ex) {
+            echo "ERROR - " . $ex->getMessage();
+            exit(1);
+        }
+    }
+
+    echo "Sanitize done." . PHP_EOL;
+} else {
+    exit(0);
+}
+
+###############################
+#        COMMON FUNCTION      #
+###############################
+
+/**
+ * Undocumented function
+ *
+ * @param string $question
+ * @param boolean $hidden
+ * @return string
+ */
+function askQuestion(string $question, $hidden = false): string
+{
+    if ($hidden) {
+        system("stty -echo");
+    }
+    printf("%s", $question);
+    $handle = fopen("php://stdin", "r");
+    $response = trim(fgets($handle));
+    fclose($handle);
+    if ($hidden) {
+        system("stty echo");
+    }
+    printf("\n");
+    return $response;
+}
+
+/**
+ * Display usage message.
+ *
+ */
+function usage(): void
+{
+    echo "This script will sanitize all your svg files and replace your corrupted images by a generic image. \n";
+}
+
+/**
+ * Check MIME Type of file
+ *
+ * @param array<string> $files
+ * @return array<string>
+ */
+function getInvalidImages(array $files): array
+{
+    $mimeTypeFileExtensionConcordance = [
+        "svg" => "image/svg+xml",
+        "jpg" => "image/jpeg",
+        "jpeg" => "image/jpeg",
+        "gif" => "image/gif",
+        "png" => "image/png"
+    ];
+    $invalidImages = [];
+    foreach ($files as $file) {
+        $fileExploded = explode(".", $file);
+        $fileExtension = end($fileExploded);
+        $mimeType = mime_content_type($file);
+        /**
+         * If MIME type is invalid or extension doesn't match MIME type
+         */
+        if (
+            !preg_match('/(^image\/(jpg|jpeg|svg\+xml|gif|png)$)/', $mimeType)
+            || (preg_match('/^image\//', $mimeType) && $mimeType !== $mimeTypeFileExtensionConcordance[$fileExtension])
```

Comment:
`$mimeTypeFileExtensionConcordance[$fileExtension]` will generate a NOTICE message if the content of `$fileExtension` is not listed in `$mimeTypeFileExtensionConcordance`